### PR TITLE
PLATFORM-1541 Fix vignette SVG image scaling

### PR DESCRIPTION
**Pull request by pchojnacki**
this PR includes fixes for bug introduced in #99 
Addtionally I added better handling of content disposition headers that might be utf-8 encoded

### CODE REVIEW

File: src/vignette/storage/s3.clj [hunk 1]
```diff
@@ -91,8 +91,9 @@
     (when-let [object (safe-get-object (add-timeouts :get (:creds this)) bucket path)]
       (when (valid-s3-get? object)
         (let [stream (:content object)
-              meta-data (:metadata object)]
-          (create-stored-object stream meta-data)))))
+              meta-data (:metadata object)
+              file-name (last (clojure.string/split (:key object) #"/"))]
```
Comment (drsnyder) [inline src/vignette/storage/s3.clj]: Could this be `nil`? What should the behavior be in that case? Should we use the hash?

**Comment by drsnyder** [comment]
How does this address the bug introduced in #99?

File: src/vignette/storage/s3.clj (hunk 1 above)
Comment (pchojnacki) [inline src/vignette/storage/s3.clj]: Good point but since :key object is the id of the object then if object is found it should have :key property set.

**Comment by pchojnacki** [comment]
#99 worked by adding original filename to UUID for temp filename but that caused problems with string escaping in bash for some cases. Since it is sufficient to only use extension for the imagick/pantomime mime type detection to work correctly.  Here I've change that code to only append extension to the uuid.

File: src/vignette/util/thumbnail.clj [hunk 2]
```diff
@@ -109,10 +110,13 @@
...
             "unable to get original for thumbnailing")))
 
+(defn- extract-extension [filename]
+  (last (split filename #"\.")))
+
 (defn original->local
   "Take the original and make it local."
   [original]
-  (let [temp-file (io/file (temp-filename))]
+  (let [temp-file (io/file (temp-filename nil (extract-extension (filename original))))]
```
Comment (drsnyder) [inline src/vignette/util/thumbnail.clj]: Ah, ok, this looks like the part that fixes the previous PR. Can you add tests for this behavior?

**Comment by drsnyder** [comment]
Minor comment, otw, LGTM.

**Pull request merged**